publisher-sdk/src/androidTest/java/com/criteo/publisher/privacy/ConsentDataTest.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.privacy
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.preference.PreferenceManager
+import com.criteo.publisher.mock.MockedDependenciesRule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import javax.inject.Inject
+
+class ConsentDataTest {
+  @Rule
+  @JvmField
+  val mockedDependenciesRule = MockedDependenciesRule()
+
+  @Inject
+  private lateinit var context: Context
+
+  private lateinit var sharedPreferences: SharedPreferences
+
+  @Before
+  fun setUp() {
+    sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
```

Comment:
The shared pref injected there https://github.com/criteo/android-publisher-sdk/pull/187/files#diff-488c51c5458586a79ec9705c7ee3d77c8b2a5b5b88dbb9879e78148f257ff704R712 is the private shared pref of the SDK.
I guess that using the SDK one is better than using the default one, to avoid colliding with the publisher.

Also, when using the SDK one in tests, you can simply put an `@Inject` annotation to get it. Also, you don't need to clean its state, the `MockedDependenciesRule` already handle it.

Reply:
> I guess that using the SDK one is better than using the default one, to avoid colliding with the publisher.

Yes, and in the public code it is the private one that is used not the public one.

But sure I can use the private one in the test as well.